### PR TITLE
CPLP-1378 updated more document type

### PR DESCRIPTION
**Pull request by qxz2mqe**
Additional changes of 12th September--support more document type for id 5,6,7
https://jira.catena-x.net/browse/CPLP-1378

### CODE REVIEW

File: src/portalbackend/CatenaX.NetworkServices.PortalBackend.PortalEntities/Enums/DocumentTypeId.cs [hunk 1]
```diff
@@ -5,5 +5,9 @@ public enum DocumentTypeId : int
     CX_FRAME_CONTRACT = 1,
     COMMERCIAL_REGISTER_EXTRACT = 2,
     APP_CONTRACT = 3,
-    DATA_CONTRACT = 4
+    DATA_CONTRACT = 4,
+    ADDITIONAL_DETAILS = 5,
+    APP_LEADIMAGE = 6,
+    APP_IMAGE = 7,
+
```
Comment (Phil91) [inline src/portalbackend/CatenaX.NetworkServices.PortalBackend.PortalEntities/Enums/DocumentTypeId.cs:12]: Remove empty line

File: src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppReleaseProcessController.cs [hunk 2]
```diff
@@ -86,7 +86,7 @@ public async Task<IActionResult> UpdateApp([FromRoute] Guid appId, [FromBody] Ap
     [Route("updateappdoc/{appId}/documentType/{documentTypeId}/documents")]
     [Authorize(Roles = "app_management")]
     [Consumes("multipart/form-data")]
-    [RequestFormLimits(ValueLengthLimit = 819200, MultipartBodyLengthLimit = 819200)]
+    //[RequestFormLimits(ValueLengthLimit = 819200, MultipartBodyLengthLimit = 819200)]
```
Comment (Phil91) [inline src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppReleaseProcessController.cs:89]: Either remove commented out line or comment in again

File: src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppReleaseBusinessLogic.cs [hunk 3]
```diff
@@ -104,9 +107,9 @@ public Task<int> CreateAppDocumentAsync(Guid appId, DocumentTypeId documentTypeI
         {
             throw new ArgumentException($"AppId must not be empty");
         }
-        if (!(documentTypeId == DocumentTypeId.APP_CONTRACT || documentTypeId == DocumentTypeId.DATA_CONTRACT))
+        if (!(_settings.DocumentTypeIds.Contains(documentTypeId)))
         {
-            throw new ArgumentException($"documentType must  be either APP_CONTRACT or DATA_CONTRACT");
+            throw new ArgumentException($"documentType must  be either :{string.Join(",", _settings.DocumentTypeIds)}");
```
Comment (Phil91) [inline src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppReleaseBusinessLogic.cs:112]: one whitespace too much between 'must' and 'be'

File: src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppReleaseBusinessLogic.cs [hunk 4]
```diff
@@ -104,9 +107,9 @@ public Task<int> CreateAppDocumentAsync(Guid appId, DocumentTypeId documentTypeI
         {
             throw new ArgumentException($"AppId must not be empty");
         }
-        if (!(documentTypeId == DocumentTypeId.APP_CONTRACT || documentTypeId == DocumentTypeId.DATA_CONTRACT))
+        if (!(_settings.DocumentTypeIds.Contains(documentTypeId)))
```
Comment (Phil91) [inline src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppReleaseBusinessLogic.cs:110]: the bracelets wrapping _settings.DocumentTypeIds.Contains(documentTypeId) can be removed in my opinion

File: src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppReleaseBusinessLogic.cs [hunk 5]
```diff
@@ -34,14 +35,16 @@ namespace CatenaX.NetworkServices.App.Service.BusinessLogic;
 public class AppReleaseBusinessLogic : IAppReleaseBusinessLogic
 {
     private readonly IPortalRepositories _portalRepositories;
-    
+    private readonly AppsSettings _settings;
```
Comment (Phil91) [inline src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppReleaseBusinessLogic.cs:38]: please add a line break after this line

**Review by Phil91 — CHANGES_REQUESTED** [review]
Since there are changes to the documentTypeId a new migration needs to be added

File: src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppReleaseBusinessLogic.cs (hunk 5 above)
Comment (qxz2mqe) [reply src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppReleaseBusinessLogic.cs:38]: fixed

File: src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppReleaseBusinessLogic.cs (hunk 4 above)
Comment (qxz2mqe) [reply src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppReleaseBusinessLogic.cs]: fixed

File: src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppReleaseBusinessLogic.cs (hunk 3 above)
Comment (qxz2mqe) [reply src/marketplace/CatenaX.NetworkServices.App.Service/BusinessLogic/AppReleaseBusinessLogic.cs]: fixed

File: src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppReleaseProcessController.cs (hunk 2 above)
Comment (qxz2mqe) [reply src/marketplace/CatenaX.NetworkServices.App.Service/Controllers/AppReleaseProcessController.cs]: fixed

**Comment by qxz2mqe** [comment]
Since there are changes to the documentTypeId a new migration needs to be added
As discussed with Julia today  this task assigned  to philip

**Review by ntruchsess — APPROVED** [review]

**Review by Phil91 — APPROVED** [review]

**Pull request merged**